src/core/server/saved_objects/import/check_conflicts.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import { v4 as uuidv4 } from 'uuid';
+import {
+  SavedObject,
+  SavedObjectsClientContract,
+  SavedObjectsImportError,
+  SavedObjectsImportRetry,
+} from '../types';
+import { ISavedObjectTypeRegistry } from '..';
+
+interface CheckConflictsOptions {
+  savedObjectsClient: SavedObjectsClientContract;
+  typeRegistry: ISavedObjectTypeRegistry;
+  namespace?: string;
+}
+
+interface GetImportIdMapForRetriesOptions {
+  typeRegistry: ISavedObjectTypeRegistry;
+  retries: SavedObjectsImportRetry[];
+}
+
+type InexactMatch<T> = {
+  object: SavedObject<T>;
+  destinations: Array<{ id: string; title?: string; updatedAt?: string }>;
+};
+type Left<T> = { tag: 'left'; value: InexactMatch<T> };
+type Right<T> = { tag: 'right'; value: SavedObject<T> };
+type Either<T> = Left<T> | Right<T>;
+const isLeft = <T>(object: Either<T>): object is Left<T> => object.tag === 'left';
+
+const createQueryTerm = (input: string) => input.replace(/\\/g, '\\\\').replace(/\"/g, '\\"');
```

Comment:
I did quite a bit of testing with this at the time -- the escaping isn't to prevent injection per se, it's to ensure that IDs that contain the `\` or `"` characters don't break the query (which would not show up as results).

My understanding is that any phrase enclosed in quotes is treated differently, if it includes "operators" those are not treated as such. If we blindly escape all characters that may be operators, we'll actually break the query for IDs that contain those characters. I'll set up a quick call with you soon to go over this and show you what I've got as far as testing is concerned.

Reply:
> My understanding is that any phrase enclosed in quotes is treated differently, if it includes "operators" those are not treated as such.

Ah, you're likely right (as usual). I didn't appreciate this when reading the docs the first time around

Reply:
> I didn't appreciate this when reading the docs the first time around

FWIW, the docs don't really spell it out. That's all I was able to recreate during my testing, though. It's definitely worth reaching out to the ES team to verify, I'll make a note to do that.